pkg/cache/memcached.go

```diff
@@ -81,3 +81,42 @@ func (c *MemcachedCache) Fetch(ctx context.Context, keys []string) map[string][]
 	c.hits.Add(float64(len(results)))
 	return results
 }
+
+// mockedMemcachedClient is a mocked memcached client for testing.
+type mockedMemcachedClient struct {
+	cache             map[string][]byte
+	mockedGetMultiErr error
+}
+
+// NewMockedMemcachedClient returns a mocked memcached client.
+func NewMockedMemcachedClient(mockedGetMultiErr error) *mockedMemcachedClient {
```

Comment:
tiny nit: maybe it's clear already that the error is mocked because it

Reply:
Also Having mock in production code is not the best idea. Let's either put it inside `_test` file or create another package.